harness/determined/cli/shell.py

```diff
@@ -130,14 +139,18 @@ def _prepare_cert_bundle(retention_dir: Union[Path, None]) -> Union[str, bool, N
 
 
 def _open_shell(
+    sess: api.Session,
     master: str,
     shell: Dict[str, Any],
     additional_opts: List[str],
     retain_keys_and_print: bool,
     print_only: bool,
 ) -> None:
+    task = bindings.get_GetTask(sess, taskId=shell["id"]).task
+    check_gt(len(task.allocations), 0, "Shell must have at least one allocation")
+    check_true(task.allocations[0].isReady, "Shell must be ready")
+
```

Comment:
can this reuse `wait_ntsc_ready`, like what `start_shell` is using when it's not detached?

this'll also make `det shell open` to wait for shell to come up. seems odd to me to have a user experience where `det shell open` errs out immediately and they have to relaunch it to retry. or did someone specifically ask for this?

Reply:
not reported or a specific ask, just a bug I found (I thought I caused somewhere else but it already existed on main) 

```wait_ntsc_ready``` feels reasonable to me if we want to change the current behavior

should we still check ready state and error before waiting at ```wait_ntsc_ready``` like Notebooks and Tensorboards do for ```open?```

should Notebooks / Tensorboards opening also call ```wait```, part of me thinks no since we send them to the UI already and that shows a loading screen of sorts while the task is RUNNING but not ready?

Reply:
changed open to now wait